api/custom-map-ui/geoname-search-ui.html

```diff
@@ -0,0 +1,452 @@
+        }
+        .result_description {
+            font-size: 11px;
+        }
+        .search {
+            height:81%;
+            float:left;
+            width: 20%;
+            max-width: 20%;
+            margin: auto;
+        }
+        .search_input {
+            width:70%;
+        }
+        .search_button {
+            width:20%;
+        }
+      </style>
+</head>
+<body>
+    
+    <div class="search" role="search">
+        <h2>GeoName Search</h2>
+
+        <input type="search" class="search_input" id="query" placeholder="Locations in Canada" autocomplete="off" name="query" size="20%"/>
+        <button onclick="update()" class="search_button btn btn-primary btn-sm" id="searchBtn">Search</button>
+        
+        <br>
+        <output id="results_count" for="query"></output>
+        <div id="results" class="results_container"></div>
+    </div>
+
+    <mapml-viewer projection="OSMTILE" zoom="4" lat="58.276703" lon="-103.710938" controls controlslist="nofullscreen" id="map">
+        <layer- label="OpenStreetMap" src="https://geogratis.gc.ca/mapml/en/osmtile/osm/" checked></layer->
+    </mapml-viewer>
+
+    <!-- ############################################### TEMPLATES ############################################### -->
+    <!-- Template layer to add to the mapml-viewer -->
+    <template id = "new_layer">
+        <layer- label="Test" checked="" id="layer">
+            <map-meta name="projection" content="OSMTILE"></map-meta>
+            <map-feature class="child" zoom="2">
+                <map-featurecaption id="caption">Bounding Box Extent</map-featurecaption>
+                <map-geometry id="geom">
+                </map-geometry>
+                <map-properties id="properties"></map-properties>
+            </map-feature>
+        </layer->
+    </template>
+
+    <!-- Template to add geometrycollection layers and its component -->
+    <template id="geometrycollection_temp">
+        <map-geometrycollection id="geometrycollection"></map-geometrycollection>
+    </template>
+
+    <!-- Templates to add features to geometry -->
+    <template id ="linestring">
+        <map-linestring>
+            <map-coordinates id="coords"></map-coordinates>
+        </map-linestring>
+    </template>
+
+    <template id="polygon">
+        <map-polygon id="poly">
+        </map-polygon>
+    </template>
+
+    <template id="point_temp">
+        <map-point id="point">
+        </map-point>
+    </template>
+
+    <!-- Template to add coordinates to features -->
+    <template id="coords_temp">
+        <map-coordinates id="coords"></map-coordinates>
+    </template>
+    <!-- ############################################### TEMPLATES ############################################### -->
+    
+    <script>
+        // Handling pressing Enter on focused search results divs
+        function handleEnter(e) {
+            var keycode = (e.keyCode ? e.keyCode : e.which);
+            if (keycode == '13') {
```

Comment:
In terms of readability it may be better to do x === Number.parseInt("13") rather than x == "13", where x is a number. But that's definitely for you to decide.